src/js/main.js

```diff
@@ -31,12 +31,13 @@ var testSuites = [];
 var testFilters = [];
 var currentTest;
 
+// Populate the device selection drop down menu after the gUM dialog closes.
 document.querySelector('gum-dialog').addEventListener('closed', function() {
-  if (typeof MediaStreamTrack.getSources === 'undefined') {
-    console.log('getSources is not supported, device selection not possible.');
-  } else {
-    MediaStreamTrack.getSources(gotSources);
-  }
+  navigator.mediaDevices.enumerateDevices()
+  .then(gotSources)
+  .catch(function(err) {
+      console.log('JS Device selection not supported', err);});
```

Comment:
this syntax looks strange... i would expect a new line after console.log(); and before });